process/block/preprocess/basePreProcess.go

```diff
@@ -270,6 +270,33 @@ func (bpp *basePreProcess) saveTransactionToStorage(
 			"error", errNotCritical,
 		)
 	}
+
+	if len(txInfoFromMap.tx.GetUserTransactions()) == 0 {
+		return
+	}
+
+	for _, innerTx := range txInfoFromMap.tx.GetUserTransactions() {
+		innerTxHash, errCalculateHash := core.CalculateHash(bpp.marshalizer, bpp.hasher, innerTx)
```

Comment:
can you extract this into a method saveInnerTxToStorage

Reply:
done